src/systems/filecoin_blockchain/struct/block/block.id

```diff
@@ -0,0 +1,56 @@
+import ipld "github.com/filecoin-project/specs/libraries/ipld"
+import filcrypto "github.com/filecoin-project/specs/libraries/filcrypto"
+
+import st "github.com/filecoin-project/specs/systems/filecoin_vm/state_tree"
+import clock "github.com/filecoin-project/specs/systems/filecoin_nodes/clock"
+import addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
+import msg "github.com/filecoin-project/specs/systems/filecoin_vm/message"
+
+type ChainWeight UVarint
+type ChainEpoch UVarint
+
+type BlockCID ipld.CID
+type MessageRoot ipld.CID
+type ReceiptRoot ipld.CID
+
+type BlockHeader struct {
+    // Chain linking
+    Parents          Tipset
+    Weight           ChainWeight
+    Epoch            ChainEpoch
+
+    // Miner Info
+    MinerAddress     addr.Address
+
+    // State
+    StateTree        st.StateTree
+    Messages         MessageRoot
```

Comment:
Putting messages here confuses the parent state context of the state tree and receipts. The latter two are the state, the messages is a proposal for the next tipset.

Reply:
Agree! good catch thanks will reorganize